### PR TITLE
Move suitesparse to stdlib

### PR DESCRIPTION
**Pull request by ViralBShah**
As part of JuliaLang/LinearAlgebra.jl#446, move suitesparse to stdlib.

I can't quite figure out how to hook up the stdlib/SuiteSparse to base/sysimg.jl and the tests. Hence, if anyone can suggest what to do - I can do that, or commits on this branch appreciated.

### CODE REVIEW

File: base/precompile.jl [hunk 1]
```diff
@@ -10,9 +10,9 @@
 
 precompile(Tuple{typeof(Base.pointer), Array{UInt8, 1}, UInt64})
 precompile(Tuple{typeof(Base.convert), Type{Ptr{Int32}}, Ptr{UInt8}})
-if USE_GPL_LIBS
-precompile(Tuple{typeof(Base.SparseArrays.CHOLMOD.set_print_level), Array{UInt8, 1}, Int64})
-end
+#if USE_GPL_LIBS
+#precompile(Tuple{typeof(Base.SparseArrays.CHOLMOD.set_print_level), Array{UInt8, 1}, Int64})
+#end
```
Comment (JeffBezanson) [inline base/precompile.jl]: This can just be deleted.

**Comment by ViralBShah** [comment]
Should be good to merge if we get a green on CI. Rebased over the recent sparse changes. Would be nice to get it in before further sparse changes if possible.

**Comment by KristofferC** [comment]
Need to add `:SuiteSparse` to the test in `compile.jl` here: https://github.com/JuliaLang/julia/blob/fb2ba438c6b45327bcd2d4d062d5f9e26284ec98/test/compile.jl#L221

**Comment by ViralBShah** [comment]
Failing test was unrelated.

**Comment by ararslan** [comment]
Thanks for doing this, Viral!

**Pull request merged**